### PR TITLE
Fix: response nil attachments

### PR DESCRIPTION
**Pull request by fangyincheng**
- fix bug about attachments
- update some todo

### CODE REVIEW

File: hessian.go [hunk 1]
```diff
@@ -43,6 +43,8 @@ type DubboHeader struct {
 	ID             int64
 	BodyLen        int
 	ResponseStatus byte
+
+	Attachments map[string]string // attr
```
Comment (gaoxinge) [inline hessian.go]: `Attachments`作为`Service`的一部分是不是更加合理一点，而不是作为`DubboHeader`的一部分？

Comment (fangyincheng) [reply hessian.go]: attachments是一些附带的信息，并不属于一个service服务的属性，所以我放在了header，不过这部分我觉得 #46 fix的时候需要根据java的dubbo协议再整理一下

Comment (gaoxinge) [reply hessian.go]: @fangyincheng 嗯，你说的对的：attachments不属于一个service服务的属性。但attchment作为附加信息，最好不要放到header里面，最好放到body里面。

Comment (fangyincheng) [reply hessian.go]: - 这里虽然attachments在readbody里面decode，但是其实它并不是response或者request的主体，我觉得它类似于http头部的附带字段，而且刚好是个map；不过因为它在dubbo协议里面是在body内容后面，所以我觉得暂时放header和body都可以吧。

- 这里的header和body是不是可以重新定义下结构体呢？比如Request和Response

**Comment by wongoo** [comment]
@fangyincheng  As [dubbo implement](http://dubbo.apache.org/zh-cn/docs/dev/implementation.html) says , the `attachment` is a part of `Variable Part` for `Request` message, so it should not be added into dubbo header. Instead a standard `Request` struct can be defined.

**Comment by wongoo** [comment]
@fangyincheng I think supporting additional feature should be an improvement, not a bug. If it's an improvement , an issue is prefer to be submitted first, so that the communities can do more discuss about it .

**Comment by gaoxinge** [comment]
@wongoo @fangyincheng As you say, It's a good idea to define a `Request` and `Response` struct.

**Comment by fangyincheng** [comment]
@wongoo 
I think: 
- This pr consists of two parts. One is that a logic error is fixed in the code, the other is support for attachments. In fact, I've coded part of it before, and the pr has perfect it.
- if we apply "define a Request and Response struct", `ReadHeader` and `ReadBody` will be modified. My problem is whether we should more modify call of hessian now. Or delay it?

File: response.go [hunk 2]
```diff
@@ -61,18 +62,22 @@ func packResponse(header DubboHeader, attachments map[string]string, ret interfa
...
-		atta := isSupportResponseAttachment(attachments[DUBBO_VERSION_KEY])
+		var atta bool
+		if header.Attachments == nil {
+			atta = false
+		} else {
+			atta = isSupportResponseAttachment(header.Attachments[DUBBO_VERSION_KEY])
+		}
 
 		var resWithException, resValue, resNullValue int32
 		if atta {
```
Comment (u0x01) [inline response.go]: there is bug of dubbo 2.5.x supporting here

**Comment by u0x01** [comment]
> @fangyincheng I think supporting additional feature should be an improvement, not a bug. If it's an improvement , an issue is prefer to be submitted first, so that the communities can do more discuss about it .

https://github.com/dubbogo/hessian2/pull/47/files/83950c576c04cdca9a6c09459ad057669b652a7d#r288378148

**Pull request merged**